includes/collection/class-remote-actors.php

```diff
@@ -625,4 +634,35 @@ public static function get_acct( $id ) {
 
 		return $acct;
 	}
+
+	/**
+	 * Get the avatar URL for a remote actor.
+	 *
+	 * @param int $id The ID of the remote actor post.
+	 *
+	 * @return string The avatar URL or empty string if not found.
+	 */
+	public static function get_avatar_url( $id ) {
+		$avatar_url = \get_post_meta( $id, '_activitypub_avatar_url', true );
+		if ( $avatar_url ) {
+			return $avatar_url;
+		}
+
+		// If not found in meta, try to extract from post_content JSON.
+		$post = \get_post( $id );
+		if ( ! $post || empty( $post->post_content ) ) {
+			return '';
+		}
+
+		$actor_data = \json_decode( $post->post_content, true );
+		if ( empty( $actor_data['icon']['url'] ) ) {
+			return '';
+		}
+
+		$avatar_url = $actor_data['icon']['url'];
```

Comment:
```suggestion
		$avatar_url = object_to_uri( $actor_data['icon'] );
```
In case `icon` could be a URL.